### PR TITLE
Bug 1549471 - Optimize ds cache loads and saves

### PR DESCRIPTION
**Pull request by ScottDowne**

### CODE REVIEW

File: lib/DiscoveryStreamFeed.jsm [hunk 1]
```diff
@@ -36,13 +38,25 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
   constructor() {
     // Internal state for checking if we've intialized all our data
     this.loaded = false;
-
-    // Persistent cache for remote endpoint data.
-    this.cache = new PersistentCache(CACHE_KEY, true);
+    this._kvstore = KeyValueService.getOrCreate(OS.Constants.Path.localProfileDir, CACHE_KEY);
```
Comment (ncloudioj) [inline lib/DiscoveryStreamFeed.jsm]: Note that each rkv store supports multiple sub-stores, and the signature of `KeyValueService.getOrCreate(store_path, store_name)`, So you might want to add one more level to the `store_path`. For example, say we want to create a store for "discovery-stream" with a sub-store named "remote-content", 
```js
const path = OS.Path.join(
  OS.Constants.Path.localProfileDir,
  "discovery-stream"
);
this._kvstore = KeyValueService.getOrCreate(path, "remote-content");

// You can also create another sub-store, say "foo", in the same store
this._store = KeyValueService.getOrCreate(path, "foo");
```

See its API doc [here](https://searchfox.org/mozilla-central/source/toolkit/components/kvstore/nsIKeyValue.idl#45).

File: lib/DiscoveryStreamFeed.jsm [hunk 2]
```diff
@@ -189,12 +203,13 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
   }
 
   async _checkExpirationPerComponent() {
-    const cachedData = await this.cache.get() || {};
-    const {feeds} = cachedData;
+    const layout = await this.kvstoreGet("layout") || {};
```
Comment (ncloudioj) [inline lib/DiscoveryStreamFeed.jsm]: How many key/value pairs do we have in this store? If there are only three of them we can just load all of them in one call using the [enumerator](https://searchfox.org/mozilla-central/source/toolkit/components/kvstore/kvstore.jsm#173).

File: lib/DiscoveryStreamFeed.jsm (hunk 1 above)
Comment (ScottDowne) [reply lib/DiscoveryStreamFeed.jsm]: Is there performance implications around loading something out of 1 store if it's got a lot of data, vs splitting the data into two smaller stores, and only loading it from the smaller store as needed?

I was under the impression that the size of the store wasn't significant.

**Comment by ScottDowne** [comment]
I don't think there are performance gains to be had here.

I'm going to close it for now.

**Pull request closed without merging**